### PR TITLE
Feature/152577745 analytics endpoint

### PR DESCRIPTION
**Pull request by yaelheed**

### CODE REVIEW

File: service/src/main/java/com/heed/job/dispatcher/job/JobRequest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+@Getter
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@EqualsAndHashCode
+@ToString
+public class JobRequest {
+
+    String id;
```
Comment (zvikai) [inline service/src/main/java/com/heed/job/dispatcher/job/JobRequest.java]: private

File: service/src/main/java/com/heed/job/dispatcher/job/JobStatusAspect.java [hunk 2]
```diff
@@ -1,44 +1,63 @@
...
-        jobStatusMessageSender.send(statusMessage.toJson());
+    @Around("execution(void com.heed.job.dispatcher.job.JobDispatcher.dispatch(..))")
+    public void updateOnDispatchFailureSwallowException(ProceedingJoinPoint joinPoint) {
+        Job job = (Job) joinPoint.getArgs()[0];
+        try{
+            joinPoint.proceed();
+            log.info("After Job Dispatch");
+            send(createStatusMessage(job.getClipId(), JobStatus.IN_PROGRESS, 30));
+
+        } catch (Throwable ex) {
```
Comment (zvikai) [inline service/src/main/java/com/heed/job/dispatcher/job/JobStatusAspect.java]: not good practice catch throwable

File: service/src/main/java/com/heed/job/dispatcher/job/JobStatusAspect.java [hunk 3]
```diff
@@ -1,44 +1,63 @@
...
-    public void updateJobStatusAfterJobDispatch(JobDetails jobDetails) {
-        JobStatusMessage statusMessage = JobStatusMessage.builder()
-                .id(jobDetails.getClipId())
-                .status(JobStatusMessage.Status.builder().code(JobStatus.IN_PROGRESS_PENDING_RENDERING.value()).percentage(30).build())
-                .build();
-        jobStatusMessageSender.send(statusMessage.toJson());
+    @Around("execution(void com.heed.job.dispatcher.job.JobDispatcher.dispatch(..))")
+    public void updateOnDispatchFailureSwallowException(ProceedingJoinPoint joinPoint) {
+        Job job = (Job) joinPoint.getArgs()[0];
+        try{
```
Comment (zvikai) [inline service/src/main/java/com/heed/job/dispatcher/job/JobStatusAspect.java]: space

File: service/adapter_stubs/ct_adapter.py [hunk 4]
```diff
@@ -1,3 +1,7 @@
-import sys
-sys.stdout.write(sys.stdin.read())
+import requests, sys, json
```
[the remark this answers is not in the document]
Comment (yaelheed) [reply service/adapter_stubs/ct_adapter.py]: I have a test that calls make in job-dispatcher which calls the python and then the python calls the fetch analytics endpoint in job-dispatcher. I then verify that I get a call to the mock analytics server.
This mocks the sdk for sending the analytics request to job-dispatcher.

File: service/src/main/java/com/heed/job/dispatcher/common/SystemTimeProvider.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package com.heed.job.dispatcher.common;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class SystemTimeProvider implements TimeProvider {
```
[the remark this answers is not in the document]
Comment (yaelheed) [reply service/src/main/java/com/heed/job/dispatcher/common/SystemTimeProvider.java]: You're right I removed the class

File: service/src/main/java/com/heed/job/dispatcher/common/TimeConversion.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package com.heed.job.dispatcher.common;
+
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+public class TimeConversion {
```
[the remark this answers is not in the document]
Comment (yaelheed) [reply service/src/main/java/com/heed/job/dispatcher/common/TimeConversion.java]: This also was removed

File: service/src/main/java/com/heed/job/dispatcher/job/JobStatusAspect.java (hunk 3 above)
Comment (yaelheed) [reply service/src/main/java/com/heed/job/dispatcher/job/JobStatusAspect.java]: done

File: service/src/main/java/com/heed/job/dispatcher/job/JobStatusAspect.java (hunk 2 above)
Comment (yaelheed) [reply service/src/main/java/com/heed/job/dispatcher/job/JobStatusAspect.java]: This si what the aspect method requires, I can't change it to exception

File: service/src/main/java/com/heed/job/dispatcher/analytics/AnalyticsRequester.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package com.heed.job.dispatcher.analytics;
+
+public interface AnalyticsRequester {
+
+    void requestAnalyticsFor(AnalyticsQuery query) throws AnalyticRequesterException;
```
[the remark this answers is not in the document]
Comment (yaelheed) [reply service/src/main/java/com/heed/job/dispatcher/analytics/AnalyticsRequester.java]: I took it out...

**Pull request merged**